parser/src/test/java/com/microsoft/gctoolkit/parser/ConcurrentMarkSweepPhaseParserRulesTest.java

```diff
@@ -30,26 +33,26 @@ public void testCMSParseRules() {
 
     /* Code that is useful when testing individual records */
 
-    private final boolean debugging = Boolean.getBoolean("microsoft.debug");
-
     @Test
     public void testDebugCMSParseRules() {
         int index = 0;
         GCParseRule rule = rules[index];
-        evaluate(rule, lines[index], debugging);
+        evaluate(rule, lines[index]);
     }
 
 
-    private void evaluate(GCParseRule rule, String string, boolean dump) {
+    private void evaluate(GCParseRule rule, String string) {
 
         GCLogTrace trace = rule.parse(string);
         assertNotNull(trace);
-        if (dump) {
-            LOGGER.fine("matches groups " + trace.groupCount());
+        // Enable debugging by setting gctoolkit.debug to true
+        GCToolKit.LOG_DEBUG_MESSAGE(() -> {
+            StringBuilder sb = new StringBuilder("matches groups " + trace.groupCount());
             for (int i = 0; i <= trace.groupCount(); i++) {
-                LOGGER.fine(i + ": " + trace.getGroup(i));
+                sb.append(System.lineSeparator()).append(i).append(": ").append(trace.getGroup(i));
```

Comment:
possibly better to use a Joiner

Reply:
Yeah, that didn't work out so well because of the loop index. Went with this instead:
`sb.append(String.format("%n%d : %s", i, trace.getGroup(i))) `